cpp/src/arrow/dataset/test_util.h

```diff
@@ -157,7 +157,7 @@ class DatasetFixtureMixin : public ::testing::Test {
     std::shared_ptr<RecordBatch> lhs;
     ASSERT_OK(expected->ReadNext(&lhs));
     EXPECT_NE(lhs, nullptr);
-    AssertBatchesEqual(*lhs, batch);
+    AssertBatchesEqual(*lhs, batch, true);
```

Comment:
This feels as a bug to me that this is needed, since that flag is about checking schema _metadata_, not about checking the schema itself. I opened https://issues.apache.org/jira/browse/ARROW-18087 (we can probably handle this in a follow-up)

Reply:
I can remove this then?

Reply:
Not yet until https://github.com/apache/arrow/pull/14451 is merged. But will merge this one, and then remove it in the other PR.